source/clear-linux/tutorials/kubernetes.rst

```diff
@@ -0,0 +1,324 @@
+(Open Container Initiative)` guidelines and work seamlessly with Kubernetes.
+`Kata Containers`_\* provide strong isolation for untrusted workloads or
+multi-tenant scenarios.  Runc and Kata Containers can be allocated on a
+per-pod basis so you can mix and match both on the same host to suit your needs.
+
+Prerequisites
+*************
+
+This tutorial assumes you have installed |CL| updated to the latest release
+on your host system. You can learn about the benefits for cloud orchestration
+of having an up-to-date system in the `Software Update documentation`_.
+For detailed instructions on installing |CL| on a bare metal system, follow
+the :ref:`bare metal installation tutorial<bare-metal-install>`.
+
+Before you install any new packages, update |CL| with the following command:
+
+.. code-block:: bash
+
+   sudo swupd update
+
+Install Kubernetes and CRI runtimes
+***********************************
+
+Kubernetes and a set of supported :abbr:`CRI (Container Runtime Interface)` 
+runtimes are included in the `cloud-native-basic`_ bundle. To install the 
+framework, enter the following command:
+
+.. code-block:: bash
+
+   sudo swupd bundle-add cloud-native-basic
+
+Configure Kubernetes
+********************
+
+This tutorial uses the basic default Kubernetes configuration for simplicity.
+You must define your Kubernetes configuration according to your specific
+deployment and your security needs.
+
+#. Enable IP forwarding to avoid "preflight check" errors:
+
+   Create (or edit if it exists) the file :file:`/etc/sysctl.d/60-k8s.conf`
+   and include the following line:
+
+   .. code-block:: bash
+
+      net.ipv4.ip_forward = 1
+
+   Apply the change:
+
+   .. code-block:: bash
+
+      sudo systemctl restart systemd-sysctl
+
+#. Enable kubelet service:
+
+   .. code-block:: bash
+
+      sudo systemctl enable kubelet.service
+
+#. Disable swap using one of the following methods, either:
+
+   a) Temporarily:
+
+      .. code-block:: bash
+
+         sudo swapoff -a
+
+      .. note::
+
+         Swap will be enabled at next reboot, causing failures in
+         your cluster.
+
+   or:
+
+   b) Permanently:
+
+      Mask the swap partition:
+
+      .. code-block:: bash
+
+         sudo systemctl mask `sed -r -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps`
+         sudo swapoff -a
+
+      .. note::
+
+         On systems with limited resources, some performance degradation may
+         be observed while swap is disabled.
+         On some systems you may need to substitute the command 
+         "sudo systemctl mask `sed -r -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps`"
+         with
+         "sudo systemctl mask $(sed -n -e 's#^/dev/([0-9a-z]).#dev-\1.swap#p' /proc/swaps)"
+#. Create (or edit if it exists) the hosts file that kubernetes will read to locate
+   master's host:
+
+   .. code-block:: bash
+
+      echo "127.0.0.1 localhost `hostname`" >> /etc/hosts
+
```

Comment:
@joselamego. Latest push looks good for the localhost `hosts`. 
 
I wonder if we should document that we had to manually add a file `environment` , as in `/etc/environment` in order for the proxy settings to be recognized. This could be added to *Troubleshooting* section.  

If we assume that a proxy.conf is created for crio.service.d/ we could add something like: 

`cp /etc/systemd/system/crio.service.d/proxy.conf /etc/` 
`mv proxy.conf environment`

I'd like your opinion on whether this should be added. We're faced with whether a user could successfully troubleshoot proxy issues if we don't add this information.

Reply:
@mvincerx: this has been a recurrent item for discussion. Some folks argue that proxy setting is out of the Clear Linux documentation scope, and I agree to some extend. However, I can foresee it will be a typical cause of failure, so my suggestion is to only add to the proxy configuration section the general way to verify that in the user's current terminal the proxies are appropriately set: to run `echo $http_proxy`, `echo $https_proxy` and `echo $no_proxy` and the warning that if the output is different than what is expected, the procedure will fail.
I've confirmed that the appropriate file to set the proxies in a CL system is `/etc/profile`. So, in your suggested approach, it should be `mv proxy.conf profile`
Thoughts?

Reply:
@joselamego  Yes, I'm aware that whether to include proxy config information is frequently debated. 
I really like your suggestion of adding commands like `echo $http_proxy`, etc., and of explaining that " if the output is different than what is expected, the procedure will fail."  Will you please add this in the Troubleshooting section?
While it may seem elementary, if adding it supports our users, everyone wins.

Reply:
I approve of this. @joselamego Thanks for your dedication to improving this tutorial to address potential errors that users may encounter.  I invite those who were reviewers on #248 to do one final review of #255.   

@mythi, @eadamsintel, @tpleavitt 
Please let us know if any further changes are needed by 16:30 PST today.
Note: I will follow up #255 with edits to clearly distinguish workflows: 
`k8s + Docker` vs. ` k8s +CRIO`